app/models/business_unit.rb

```diff
@@ -85,4 +92,66 @@ def private_office?
   def child_type
     "users"
   end
+
+  # usage:
+  #   set_correspondence_type_roles(category_abbreviation: 'SAR', roles: %w{ edit manage view respond })
+  # #
+  def set_category_roles(category_abbreviation:, roles:)
+    cat = Category.find_by_abbreviation!(category_abbreviation.upcase)
+    tcr = TeamCategoryRole.find_by(team_id: id, category_id: cat.id)
+    if tcr.nil?
+      category_roles << TeamCategoryRole.new_for(team: self, category: cat, roles: roles)
+    else
+      tcr.update_roles(roles)
+    end
+    save
+  end
+
+  # returns an array of category records
+  def categories
+    Category.where(id: category_roles.map(&:category_id))
```

Comment:
Do you need an AREL here? The comment suggests you only want an array and I can't see any uses that would require an AREL, so this would also work:

```
category_roles.map &:category
```